### PR TITLE
added -J option to exclude collector peers by ASN

### PR DESCRIPTION
**Pull request by albertodainotti**
This filter allows the user to exclude collector peers specified by their ASN

### CODE REVIEW

**Review by alistairking — CHANGES_REQUESTED** [review]
Couple minor nits here, but otherwise it seems fine.

At a high level I would prefer adding logic so that we could negate existing filters rather than explicitly adding them like this.
I.e., it wouldn't ever be useful to provide `-j 1234 -J 5678` (right?) so it'd be nice to be able to repurpose the existing peer ASN filtering machinery instead.

File: lib/datainterfaces/bsdi_broker.c [hunk 1]
```diff
@@ -696,6 +696,20 @@ static int update_query_url(bsdi_t *di)
       APPEND_STR(f);
     }
   }
+  // not peer asns
```
[the remark this answers is not in the document]
Comment (albertodainotti) [reply lib/datainterfaces/bsdi_broker.c:699]: Oops.

File: tools/bgpreader.c [hunk 2]
```diff
@@ -129,6 +129,9 @@ static struct bs_options_t bs_opts[] =
   {{"peer-asn", required_argument, 0, 'j'},
    "<peer ASN>",
    "return elems received by a given peer ASN*"},
+  {{"peer-asn", required_argument, 0, 'J'},
```
[the remark this answers is not in the document]
Comment (albertodainotti) [reply tools/bgpreader.c:132]: Oops again

**Review by alistairking — APPROVED** [review]
lgtm

**Pull request merged**